docs/api-reference/cpp-api.rst

```diff
@@ -17,12 +17,136 @@ To search an API, refer to the API :ref:`genindex`.
 
 Device functions
 ================
+
+To use the device API, include the file ``rocrand_kernel.h`` in files that define kernels that use rocRAND device functions. The typical usage of device functions consists of the following operations in the device kernel definition:
+
+1. Create a new generator state object of the desired generator type.
+
+2. Initialize the generator state parameters using ``rocrand_init``.
+
+3. Generate random numbers by calling generation function on the generator state.
+
+4. Use the results.
+
+Since the rocRAND device functions are invoked from inside the user kernel, the generated numbers may be used right away in the kernel without a need to copy them to the host memory.
+
+In the below example, random number generation is using XORWOW generator.
+
+..  code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <rocrand/rocrand_kernel.h>
+    
+    __global__
+    void test()
+    {
+        uint                 tid = blockDim.x * blockIdx.x + threadIdx.x;
+        rocrand_state_xorwow state;
+        rocrand_init(123, tid, 0, &state);
+    
+        for(int i = 0; i < 3; ++i)
+        {
+            const auto value = rocrand(&state);
+            printf("thread %d, index %u: %u\n", tid, i, value);
+        }
+    }
+    
+    int main()
+    {
+        test<<<dim3(1), dim3(32)>>>();
+        hipDeviceSynchronize();
+    }
+
 .. doxygengroup:: rocranddevice
 
 C host API
 ==========
+
+C host API allows encapsulation of the internal generator state. Random numbers may be produced either on the host CPU or device GPU, whether an appropriate generator object was created. The typical sequence of operations for GPU generation consists of the following steps:
+
+1. Allocate memory on the device with ``hipMalloc``.
+
+2. Create a new generator of the desired type with ``rocrand_create_generator``.
+
+3. Set the generator options, for example, use ``rocrand_set_seed`` to set the seed.
+
+4. Generate random numbers with ``rocrand_generate`` or another generation function.
+
+5. Use the results.
+
+6. Clean up with ``rocrand_destroy_generator`` and ``hipFree``.
+
+To generate random numbers on the host CPU, the memory allocation in step one should be made by host memory allocation call, in step two ``rocrand_create_generator_host`` should be called respectfully, in the last step appropriate memory release should be made beside the ``rocrand_destroy_generator``. All other calls work identically whether you are generating random numbers on the device or on the host CPU. 
```

Comment:
I don't think respectfully is the right word here. Do you mean similarly?

Reply:
Changed it in https://github.com/ROCm/rocRAND/pull/574/commits/9f8275c66b77a353c7c3cb51b6509b7cc009625b to use "instead" as it signifies that a call should be replaced.